### DISCUSSION TITLE
conditional insert

### DISCUSSION THREAD

**Discussion by andyczerwonka** [slick/slick#2256] [Questions] [answered]
I have a simple insert that looks like:
```scala
records += Record(c1, c2, c3)
```
I'd like to change it to make that insert conditional. I don't want to proceed with the insert if a record exists where the all the columns match but c3 is null.

Basically, I'd like to have Slick generate the following:
```sql
insert into mytable (c1, c2, c3) values ('one', 'two', 'three')
where not exists (select 1 from mytable where c1 = 'one' and c2 is null);
```
Can anyone point me to a solution?

**Comment by nafg** [comment]
The DSL does not currently support that. You can use Plain SQL queries instead; see https://scala-slick.org/doc/3.3.3/sql.html

**Reply by andyczerwonka** [reply]
Can you point me to an example?

**Reply by nafg** [reply]
https://scala-slick.org/doc/3.3.3/sql.html

**Reply by nafg** [reply]
I edited into my original answer.

**Reply by andyczerwonka** [reply]
Looks like there might be some DSL support, Reading [Essential Slick, section 3.1.7 More Control over Inserts](https://books.underscore.io/essential-slick/essential-slick-3.html), there seems to be a way to pull it together.

**Reply by nafg** [reply]
Wow, I did not know about that!

**Reply by nafg** [reply]
I don't think it matters (this is not stackoverflow!) but how about adding that (or the resulting code) as a separate answer?

**Reply by andyczerwonka** [reply]
I'll do that once I pull it together. 👍🏻

**Comment by nafg** [comment, marked as answer]
See [Essential Slick, section 3.1.7 More Control over Inserts](https://books.underscore.io/essential-slick/essential-slick-3.html)
